core/App/ehsm_napi.cpp

```diff
@@ -224,41 +270,78 @@ char* NAPI_Decrypt(const char* cmk_base64,
         const char* ciphertext_base64,
         const char* aad)
 {
-    string decode_cmk;
-    string decode_cipher;
-    string plaintext_base64;
-
     RetJsonObj retJsonObj;
-    ehsm_status_t ret = EH_OK;
-
-    ehsm_keyblob_t masterkey;
+    if (cmk_base64 == NULL || ciphertext_base64 == NULL) {
+        retJsonObj.setCode(retJsonObj.CODE_BAD_REQUEST);
+        retJsonObj.setMessage("paramter invalid.");
+        return retJsonObj.toChar();
+    }
+    if(aad == NULL){
+        aad = "";
+    }
 
+    ehsm_status_t ret = EH_OK;
+    ehsm_keyblob_t cmk;
     ehsm_data_t plaint_data;
     ehsm_data_t aad_data;
     ehsm_data_t cipher_data;
+    string plaintext_base64;
+    
+    string cmk_str;
+    string ciphertext_str;
+    int cmk_len;
+    int ciphertext_len;
+    int aad_len;
```

Comment:
seems you didn't update the codes here following yang's comments. and same with below codes.